### PR TITLE
Headphones

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Uses ytdl
since it uses goonchat output you cant lag the server with it, and theres a limiter on the music list

## Why It's Good For The Game

My ingame sound breaks when I play music on spotify :(

## Changelog
:cl:
add: Added headphones that play music off of youtube
/:cl:

### CODE REVIEW

File: code/modules/clothing/head/headphones.dm [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+	if(web_sound_url && !findtext(web_sound_url, GLOB.is_http_protocol))
+		to_chat(usr, "<span class='warning'>BLOCKED: Content URL not using http(s) protocol</span>")
+		to_chat(usr, "<span class='warning'>The media provider returned a content URL that isn't using the HTTP or HTTPS protocol</span>")
+		return
+
+	var/client/C = usr?.client
+	if(!C?.prefs)
+		return
+	if((C.prefs.toggles_sound & SOUND_MIDI) && C.chatOutput?.working && C.chatOutput.loaded)
```
Comment (Hughgent) [inline code/modules/clothing/head/headphones.dm:80]: This might not need the prefs check, because the player is actively choosing to 'listen to music'

**Comment by psykzz** [comment]
I love the idea but i dont think it can go through, we have issues already with youtube blocking us for admin use, if we give that to the players we'd likely be forever blocked.

**Comment by TiviPlus** [comment]
Don't know till we try!

**Review by TheBimmer231 — CHANGES_REQUESTED** [review]
That's some garbage honestly, Ping me, I'll help you out.

**Comment by TiviPlus** [comment]
Ytdl keeps breaking iwth this, sad

**Pull request closed without merging**